news/_posts/2017-11-30-make-back-office-modules-great-again.md

```diff
@@ -0,0 +1,66 @@
+
+### Current situation of Back Office modules (pre 1.7.3)
+
+In Back Office, for the modern pages (migrated to Symfony 2), PrestaShop developers can still use Hooks like in the old ways. But they are not allowed to override templates, forms or data returned by the controllers. This is forbidden by design, because we are removing the old `override` system.
+
+In Product page, we have introduced a specific a specific hook called `hookDisplayAdminProductsExtra` to display modules in a specific tab, but it's really painful to hook into the forms to add new fields, new buttons or new blocks of information.
+
+Reviewing the actual hooks, they are not located wisely regarding the new UX of this page and some hooks are missing to make basic additions (like adding new buttons to menus for instance).
+
+Finally, and like in 1.6, it's still really difficult to know when, where and if a hook has been called, how many times and what are the modules hooked.
+
+### PrestaShop 1.7.3, the module experience re-invented
+
+The first new feature we'd like to present is the Hooks debugger. In all modern pages of Back Office, you have access to the PrestaShop debug toolbar, now enriched by a "Hooks" section.
+
+![PrestaShop Hooks Debugger section](/assets/images/2017/11/hooks_debugger_section.png)
+
+At first look, in every modern page you can know which and how many times hooks have been called. If you access the section panel, you have access to many additional information:
+
+* modules and hooks arguments
+* called and uncalled (but available) hooks
+* which modules react to hooks (the order is respected)
+
+![PrestaShop Hooks Debugger](/assets/images/2017/11/hooks_debugger.png)
+
+This allow every PrestaShop developer to choice wisely the right hook, but also to understand some behaviors if a specific hook doesn't render the expected view without have to enable/disable every module until find the guilty one.
+
+Then, we have introduced two hooks in Product Catalog page to make the contribution of menus simple:
+
+* `displayDashboardToolbarTopMenu` to be able to add new action to the top menu
+* `displayDashboardToolbarIcons` to allow the addition of new icons in Dashboard block (will be available in every Dashboard modern pages)
+
+![New Hooks](/assets/images/2017/11/new_hooks.png)
+
+One more think with PrestaShop 1.7.3, the introduction of **"modules as bundle" system**, allowing the override of templates, services (classes), commands, forms and everything from modern pages in your modules. We are documenting this new system right now, but regarding the templating override we can tell this will work mostly the same way we already work with override in Front Office theme templates.
+
+### PrestaShop 1.7.4 and beyond?
+
+PrestaShop 1.7.4 will use Symfony 3 and will require PHP 5.6, we have decided to stop the maintenance of unmaintained PHP versions for security reason: it's time to upgrade PHP, ideally to the latest version (7.1 at this moment).
```

Comment:
`+At least` PHP 5.6?

Reply:
I'm not sure we will require PHP 7 for 1.7.4, what do you think @toutantic ?

Reply:
I meant "At least PHP 5.6". One version among 5.6, 7, or 7.1 is required.